src/persist-client/src/cfg.rs

```diff
@@ -0,0 +1,278 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+#![allow(missing_docs)]
+
+//! The tunable knobs for persist.
+
+use std::time::Duration;
+
+use mz_build_info::BuildInfo;
+use mz_ore::cast::CastFrom;
+use mz_ore::now::NowFn;
+use mz_persist::cfg::ConsensusKnobs;
+use mz_proto::{ProtoType, RustType, TryFromProtoError};
+use proptest_derive::Arbitrary;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+
+include!(concat!(env!("OUT_DIR"), "/mz_persist_client.cfg.rs"));
+
+/// The tunable knobs for persist.
+#[derive(Debug, Clone)]
+pub struct PersistConfig {
```

Comment:
Can you explain the relationship between `PersistConfig` and `PersistParameters`? At first glance they seem to be serving the same purpose (configuration parameter containers), so I'm not sure why we need both.

Reply:
`PersistConfig` includes all config parameters for the persist client, while `PersistParameters` only contains the dynamically configurable ones (it mirrors `ComputeParameters` and `StorageParameters`). `PersistParameters` is part of the compute and the storage protocol, and therefore has (or might have in the future) stricter backwards-compatibility requirements than `PersistConfig`. I think this is a good reason to keep them separate. It's also nice that we can encode which parameters can be changed dynamically in the type system.

Reply:
Thanks for clarifying. Maybe the relationship should be stated somewhere in the module docs. Another option is to use a more descriptive name `PersistConfig` / `DynamicPersistConfig` vs `PersistParameters`.

Reply:
yeah, Jan is correct on all counts. this relationship is much more obvious after the next PR where I actually hook them together, but adding additional commentary can't hurt. done!

as for naming, I think the name that makes the least sense to me within the persist context is PersistParameters, which might more precisely be something like PersistConfigUpdates. however, I think the current naming clearly establishes the relationship between PersistParamaters and ComputeParamaters/StorageParameters, which, since it's crossing abstraction boundaries, seems more important. as a result, I'd prefer to leave things as-is